utils/utils.go

```diff
@@ -75,3 +95,161 @@ func IsValidEthereumAddress(address string) bool {
 	re := regexp.MustCompile("^0x[0-9a-fA-F]{40}$")
 	return re.MatchString(address)
 }
+
+func ReadPublicUrl(url string) ([]byte, error) {
+	// allow no redirects
+	httpClient := http.Client{
```

Comment:
This stops the client from following re-directs

Reply:
yeah!